### PR TITLE
Introduce Payment Methods

### PR DESCRIPTION
**Pull request by oleganza**

### CODE REVIEW

**Comment by oleganza** [comment]
@devongundry @Flavien please take a look. This is a squashed version of what we have discussed 2 days ago.

File: payment-methods.mediawiki [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+[[payment-requests.mediawiki|Payment requests]] help communicate exact transaction requirements to the client. 
+The payment response is either a failure to pay, or a complete transaction exactly satisfying the Payment Request.
+
+This specification solves a problem preceding the actual payment: how do payer and payee negotiate which kinds of assets are acceptable for an upcoming transaction.
+[https://github.com/bitcoin/bips/blob/master/bip-0070.mediawiki BIP70] does not address this problem because it deals with only one kind of asset, Bitcoin.
+
+We introduce three kinds of new messages (using Protocol Buffers encoding like in [https://github.com/bitcoin/bips/blob/master/bip-0070.mediawiki BIP70]): 
+
+* '''Payment Method Request''' contains information about acceptable payment options that payer may use. Merchant sends this message to a customer.
+* '''Payment Method''' contains exact payment options (kinds and amounts of assets) that the payer is willing to use. Customer sends this as a reply to a merchant. In response, merchant either replies with a finalized [[Payment Request|payment-requests.mediawiki]] or with a Payment Method Rejection message.
```
Comment (dexX7) [inline payment-methods.mediawiki]: > [[Payment Request|payment-requests.mediawiki]]

Bad link.

**Comment by oleganza** [comment]
@dexX7 thanks for verifying it.

**Comment by Flavien** [comment]
Thanks Oleg, that looks good to me.

**Pull request merged**